src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/assembly_of_iron.cpp

```diff
@@ -110,687 +111,587 @@ enum
+                    case NPC_STEELBREAKER: instance->SetSpecialAchievementCriteria(TYPE_ACHIEV_STEELBREAKER, true); break;
+                }
+                caster->CastSpell(nullptr, SPELL_COUNCIL_KILL_CREDIT, TRIGGERED_OLD_TRIGGERED);
+                instance->SetData(TYPE_ASSEMBLY, DONE);
+                static_cast<Creature*>(caster)->SetNoLoot(false);
+            }
+            return;
         }
     }
 
-    void SpellHit(Unit* /*pCaster*/, const SpellEntry* pSpell) override
+    bool OnCheckTarget(const Spell* spell, Unit* target, SpellEffectIndex eff) const override
     {
-        // Increase the phase when hit with the supercharge spell by his brothers
-        if (pSpell->Id == SPELL_SUPERCHARGE)
+        if (eff != EFFECT_INDEX_2)
+            return true;
+        if (!target || !target->IsAlive())
+            return false;
+        if (target->GetEntry() == NPC_BRUNDIR && target->GetAuraCount(SPELL_SUPERCHARGE) == 1)
+            target->CastSpell(nullptr, SPELL_STORMSHIELD, TRIGGERED_OLD_TRIGGERED);
+        if (target->GetEntry() == NPC_STEELBREAKER && target->GetAuraCount(SPELL_SUPERCHARGE) == 1)
         {
-            // Not sure if there is a spell for this, so we are doing it here
-            m_creature->SetHealth(m_creature->GetMaxHealth());
-            ++m_uiPhase;
+            target->CastSpell(nullptr, SPELL_ELECTRICAL_CHARGE, TRIGGERED_OLD_TRIGGERED);
         }
+        target->SetHealthPercent(100.f);
+        if (target->AI())
+            target->AI()->SpellListChanged();
+        return true;
+    }
+};
 
-        if (m_uiPhase == PHASE_CHARGE_TWO)
-        {
-            // set the instace data to special in order to mark the last phase - this is used to check the achiev criteria
-            if (m_pInstance)
-                m_pInstance->SetData(TYPE_ASSEMBLY, SPECIAL);
-        }
+struct LightningTendrils : public SpellScript, public AuraScript
+{
+    SpellCastResult OnCheckCast(Spell* spell, bool strict) const override
+    {
+        Unit* caster = spell->GetCaster();
+        if (!caster)
+            return SPELL_FAILED_CASTER_DEAD;
+        if (caster->GetAuraCount(SPELL_SUPERCHARGE) < 2)
+            return SPELL_FAILED_CASTER_AURASTATE;
+        return SPELL_CAST_OK;
     }
 
-    void UpdateAI(const uint32 uiDiff) override
+    void OnApply(Aura* aura, bool apply) const override
     {
-        // Pre fight visual spell
-        if (m_uiVisualTimer)
+        if (aura->GetEffIndex() != EFFECT_INDEX_0)
+            return;
+        Unit* caster = aura->GetCaster();
+        if (!caster)
+            return;
+        if (apply)
         {
-            if (m_uiVisualTimer <= uiDiff)
+            caster->CastSpell(nullptr, SPELL_TENDRILS_VISUAL, TRIGGERED_OLD_TRIGGERED);
+            DoBroadcastText(SAY_BRUNDIR_FLY, caster);
+            caster->SetHover(true);
+            caster->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_STUNNED);
```

Comment:
already handled by the visual effect
actually requires
```cpp
caster->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_MISC_FLAGS, UNIT_BYTE1_FLAG_FLY_ANIM);
```

Reply:
If the flag is in sniff,  then add it in script.